src/pyhf/pdf.py

```diff
@@ -559,30 +509,18 @@ def nominal_rates(self):
         return self.main_model.nominal_rates
 
     def expected_actualdata(self, pars):
-        return self.main_model.expected_data(pars)
+        return self.make_pdf(pars).pdfobjs[0].expected_data()
```

Comment:
why did this get changed here? `self.main_model` still works here, or have i missed something...

Reply:
I removed logpdf from self.main_model